src/components/PopupWithConfirmation.js

```diff
@@ -1,4 +1,5 @@
 import Popup from "./Popup.js";
+import { api, catchError } from "../pages/index.js";
```

Comment:
`import { api, catchError } from "../pages/index.js";
`

Please, delete this import

From the 8th sprint it's not allowed to import anything to the classes. Only class `Popup` can be imported for extending. **`Loose coupling`** should be used: it means that you need to pass callbacks to the constructor rather than using outside elements or functions.